test/functional/apps/discover/group1/_shared_links.ts

```diff
@@ -87,25 +97,56 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
               pause: true,
             },
           });
+          if (await PageObjects.share.isShareModalOpen()) {
+            await PageObjects.share.closeShareModal();
+          }
         });
 
         it('should allow for copying the snapshot URL as a short URL', async function () {
+          // modal being open blocks test execution
+          if (await PageObjects.share.isShareModalOpen()) {
+            await PageObjects.share.closeShareModal();
+          }
           const re = new RegExp(baseUrl + '/app/r/s/.+$');
-          await PageObjects.share.checkShortenUrl();
+
           await retry.try(async () => {
-            const actualUrl = await PageObjects.share.getSharedUrl();
+            // need to close the modal for the rest of the test to execute on retries
+            if (await PageObjects.share.isShareModalOpen()) {
+              await PageObjects.share.closeShareModal();
+            }
+            if (!(await PageObjects.share.isShareMenuOpen())) {
+              await PageObjects.share.clickShareTopNavButton();
+            }
+            // do not change these testSubjects - this test is very particular
+            await testSubjects.click('Permalinks');
+            await testSubjects.setCheckbox('useShortUrl', 'check');
+            await (
+              await testSubjects.find('createShortUrl')
+            ).waitForDeletedByCssSelector('.euiLoadingSpinner');
+            const actualUrl = await testSubjects.getAttribute(
+              'copyShareUrlButton',
+              'data-share-url'
+            );
```

Comment:
I guess I'm not sure why we can't use `PageObjects.share.checkShortenUrl()` here any more, could you explain?

Reply:
This test wasn't finding the data-test-subj unless I was using testSubjects and I'm going to be doing test clean up after the UI changes again in the next iteration since the UI will be changing again.